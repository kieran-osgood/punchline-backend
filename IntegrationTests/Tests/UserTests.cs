using System.Threading.Tasks;
using GraphQL;
using GraphQL.Entities.User;
using HotChocolate;
using HotChocolate.Execution;
using IntegrationTests.Helpers;
using Snapshooter.Xunit;
using Xunit;

namespace IntegrationTests.Tests
{
    public class UserTests : IClassFixture<CustomWebApplicationFactory<Startup>>
    {
        [Fact]
        public async Task Get_User()
        {
            var factory =
                new CustomWebApplicationFactory<Startup>().WithAuthentication(TestClaimsProvider.WithUserClaims());
            var user = Utilities.GetValidAuthUser();
            var input = new UserLoginInput(user.FirebaseUid, user.Name);

            var query =
                @"mutation Login($input: UserLoginInput!) {
					login(input: $input) {
						user {
							id
							firebaseUid
							name
							jokeCount
							categories {
								id
								name
								image				
							}
						}
					}
				}";

            var executor = await factory.Services.GetRequestExecutorAsync();
            var request = QueryRequestBuilder
                .New()
                .AddAuthorizedUser()
                .SetQuery(query)
                .SetVariableValue("input", input)
                .Create();

            var result = await executor.ExecuteAsync(request);
            
            Assert.Null(result.Errors);
            
            (await result.ToJsonAsync()).MatchSnapshot();
        }
    }
}