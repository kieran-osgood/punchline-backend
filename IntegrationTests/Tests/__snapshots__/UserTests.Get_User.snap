{
  "data": {
    "login": {
      "user": {
        "id": "VXNlcgppMQ==",
        "firebaseUid": "n3mU54T2ZJTrS7DySfDtPf6dB9M2",
        "name": "Test User",
        "jokeCount": 0,
        "categories": [
          {
            "id": "Q2F0ZWdvcnkKaTQ=",
            "name": "TEST RECORD: this is a user category",
            "image": "s3:url"
          }
        ]
      }
    }
  }
}
